xtask/src/ci.rs

```diff
@@ -93,6 +97,10 @@ impl CiTask {
         )
         .run();
 
-        fmt_res.and(check_full_res).and(clippy_res).and(sort_res)
+        println!("Checking all Matrix Spec links point to same version...");
+        check_spec_links(&self.project_root.join("crates"))?;
+        println!("All links to the Matrix Spec are consistent!");
```

Comment:
Can you move the `println!`s into `check_spec_links` or remove them entirely? Also please add the `check_spec_links` result to the chain below instead of using `?` here, for consistency.

Reply:
`xtask`s error type needs some work there is no way to create one from an existing error. I fixed it but it's a bit ugly.